src/content/docs/en/upgrade-astro.mdx

```diff
@@ -0,0 +1,188 @@
+
+See the upgrade guides below for an explanation of changes, comparing the new version to the old. The upgrade guides include everything that could require you to change your own code: breaking changes, deprecations, feature removals and replacements as well as updated usage guidance. Each change to Astro includes a "What should I do?" section to help you successfully update your project code.
+
+- [Upgrade to v4](/en/guides/upgrade-to/v4/)
+- [Upgrade to v3](/en/guides/upgrade-to/v3/)
+- [Upgrade to v2](/en/guides/upgrade-to/v2/)
+- [Upgrade to v1](/en/guides/upgrade-to/v1/)
+
+### Older docs (unmaintained)
+
+Documentation for older versions of Astro is not maintained, but is available as a static snapshot. Use these versions of docs if you are unable to upgrade your project, but still wish to consult guides and reference:
+
+- [unmaintained v3.6.3 snapshot](https://docs-git-v3-docs-unmaintained-astrodotbuild.vercel.app/)
+- [unmaintained v2.10.15 snapshot](https://deploy-preview-4405--astro-docs-2.netlify.app/en/getting-started/)
+
+
+## Semantic versioning
+
+Astro attempts to adhere as much as possible to [semantic versioning](https://semver.org/), which is a set of rules developers use to determine how to assign a version number to a release. Semantic version follows predictable to inform users of the kind of changes they can expect from one version to the next.
+
+Semantic versioning enforces a pattern of `X.Y.Z` for software version numbers. These values represent **major (X)**, **minor (Y)**, and **patch (Z)** updates.
+
+### Patch changes
+
+Patch changes are the least disruptive changes. They do not change the way you use Astro, and no change to your own code is required when you update.
+
+When Astro issues a "patch" version, the last number increases. (e.g. `astro@4.3.14` -> `astro@4.3.15`)
+
+Patches may be released for reasons such as:
+
+- Internal changes that do not change Astro’s functionality:
+  - refactors
+  - performance improvements
+  - increase or change in test coverage
+  - aligning with stated documentation and expected behavior
+- Improvements to logging and error messages.
+- Re-releases after a failed release.
+
+Patch changes also include bug fixes, even in cases where users were taking advantage of existing unintended or undesirable behavior.
+
+
+### Minor changes
+
+
+Minor releases primarily introduce new features and improvements that you may wish to try, but require no changes to your code. Some existing features may also be **deprecated** (marked for deletion in a future version while continuing to function) in a minor release, giving you the opportunity to prepare for their eventual removal.
+
+Minor releases include changes such as:
+
+- **Deprecations** of existing features/options.
+- Introduction of new functionalities.
+- Upgrade of minimum required Node.js version (minor or patch), e.g. from `v18.4.0` to `v18.7.0`.
```

Comment:
Want to flag this one, I don't think we've done this in the past as a normal thing in a minor release. It's possible that we have. Before we document this as a fact I would like to know:

* Have we ever bumped the minimum Node.js version in the past and why did we do it?

My understanding of Semver is that this would be a breaking change and not allowed in a minor release; you are using `18.4.0` and Astro `4.1.0` and Astro releases `4.2.0` and now your builds break; that shouldn't happen.

If there's a very good reason to do so then I think we can make this change but would like to discuss them first.

Reply:
We have never bumped a Node version outside of a major before. The motivation for this change is that right now we're blocked by a very small amount (10-15% depending how you filter the telemetry) of users being on versions that are unsupported by Node themselves (Node only supports the latest minor of extended maintenances releases and even releases)

No production system actually runs those outdated versions, they all run the latest of every major. It's only local dev machines that are outdated (and vulnerable to security issues, since Node doesn't backport security fixes, they expect you to be on the latest of the extended maintenance). It's unlikely that those users are unable to upgrade, as even old LTS of Linux systems will have the latest 

What I suggested (somewhere... I don't remember where) is to allow ourselves to change the minimum minor of *only* the extended maintenance (For Astro 4.x, that'd be requiring to latest minor of 18) and perhaps the LTS.

Reply:
Ok, I think that's a reasonable direction to go. Would still love @bluwy's input here as well.

If we do decide on this policy, I think we should state that explicitly and give the reason why. The current text (below this) says:

> approve them only if we think that a new version of Node.js can improve the security or developer experience of our users, and impact core functionalities.

this gives us too much wiggle room in my opinion. I think this is one case where we should be more explicit to give users more confidence that we're not going to change the version for arbitrary reasons.

Reply:
I agree with that! These cases should be super explicit and hopefully rare. Not really sure how best to communicate the policy on that, but I'd hate for us to develop a reputation like TypeScript where "oh they don't actually follow SemVer"

Reply:
@matthewp and @natemoo-re, could you please provide some suggestions on how to phrase this?

Reply:
I'm also fine if we really need to bump the node version if there's a big benefit for most users. I suggested emitting warnings beforehand to help soften the impact: https://github.com/withastro/docs/pull/7396#discussion_r1530685955, but otherwise I think projects can bump the node version as long as they document this possibility beforehand.